troubleshooting-duplication/apps/movement-processor/src/main/java/com/redhat/vehicles/MovementProcessor.java

```diff
@@ -0,0 +1,196 @@
+            new VoidDeserializer(),
+            new EventDeserializer<MovementReported>(MovementReported.class));
+        producer = new KafkaProducer<>(getProducerConfig());
+
+        // TODO: initialize transactions
+
+        consumer.subscribe(Collections.singletonList(MOVEMENT_REPORTED_EVENTS_TOPIC));
+
+        while (true) {
+            ConsumerRecords<Void, MovementReported> records = consumer.poll(Duration.ofMillis(Long.MAX_VALUE));
+
+            // TODO: begin transaction
+
+            Thread.sleep(2000);
+
+            List<CompletableFuture<Float>> futures = new ArrayList<>();
+
+            for (ConsumerRecord<Void, MovementReported> record : records) {
+                MovementReported event = record.value();
+
+                futures.add(
+                    calculateSpeed(event)
+                        .whenComplete((speed, t) -> {
+                            produceResult(speed);
+                        })
+                        .exceptionally(error -> {
+                            error.printStackTrace();
+                            // TODO: abort transaction
+                            System.exit(-1);
+                            return null;
+                        })
+                );
+            }
+
+            // TODO: wait for async calculations to complete
+
+            // TODO: send offsets and commit transaction
+        }
+    }
+
+    private static void produceResult(float speed) {
+        ProducerRecord<Void, Float> result = new ProducerRecord<>(SPEED_TOPIC, speed);
+
+        Callback callback = new Callback() {
+            @Override
+            public void onCompletion(RecordMetadata metadata, Exception e) {
+                if (e != null) {
+                    logger.error(e.getMessage());
+                } else {
+                    System.out.println("Speed "  + speed + " sent to topic");
+                }
+            }
+        };
+
+        producer.send(result, callback);
+    }
+
+    private static CompletableFuture<Float> calculateSpeed(MovementReported event) {
+        return CompletableFuture.supplyAsync(() -> {
+            System.out.println("\n\nProcessing " + event);
+
+            // TODO: handle division by 0
+
+            float speed = event.distance / event.time;
+
+            return speed;
+        });
+    }
+
+    private static void waitForCompletion(List<CompletableFuture<Float>> futures) {
+        CompletableFuture[] futuresArray = futures.toArray(new CompletableFuture[futures.size()]);
+        CompletableFuture.allOf(futuresArray).join();
+    }
+
+    public static Properties getConsumerConfig() {
+
+        ClassroomConfig classroomConfig = getClassroomConfig();
+        Properties props = new Properties();
+
+        props.put(
+            ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG,
+            classroomConfig.getBoostrapServer() + ":" + classroomConfig.getBootstrapPort()
+        );
+        // TODO: change consumer group
```

Comment:
```suggestion

        // TODO: change consumer group
        
```

I think its better to put a space here since it may be hard to find